### DISCUSSION TITLE
Netboot and Remote file system

### DISCUSSION THREAD

**Discussion by izissise** [uapi-group/specifications#38] [Q&A] [answered]
Hello,
I was wondering if there is any plans regarding supporting netboot with `UKI`,
especially with remote filesystems?

For example one could setup an http server containing the rootfs and mount
this rootfs using `EFI_BLOCK_IO_PROTOCOL` (ipxe can do it using sanboot)
the efi stub could then find `EFI_BLOCK_IO_PROTOCOL` devices and pass them to the kernel.
The user would end up with a net booted kernel with a remote filesystem.

**Comment by poettering** [comment, marked as answer]
UKIs should just work on uefi/http boots afaik. (but i didn't try that myself) you can build server info into the UKI (for example, include in the kernel cmdline field). The rootfs you can either also build into the UKI (if you have a lot of RAM or a small root fs), or you can transition into the rootfs on some network share, NFS, iscsi or so.

**Reply by izissise** [reply]
Thanks a lot
